addons/binding/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/handler/JeeLinkSensorHandler.java

```diff
@@ -0,0 +1,231 @@
+                                + " (" + getThing().getUID() + ")...", th);
+                    }
+                }
+            };
+
+            JeeLinkHandler jlh = (JeeLinkHandler) getBridge().getHandler();
+            JeeLinkReadingConverter<R> c = jlh.getConverter(clazz);
+            c.addReadingListener(idStr, listener);
+            initialized = true;
+        } catch (NumberFormatException ex) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Thing has an invalid sensor id: " + idStr);
+            logger.info("JeeLink sensor handler for thing {} ({}) is OFFLINE", getThing().getLabel(),
+                    getThing().getUID());
+        }
+    }
+
+    private ScheduledFuture<?> createUpdateJob(final int updateInterval) {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                logger.trace("JeeLink sensor cyclic propagation for thing {} ({})...", getThing().getLabel(),
+                        getThing().getUID());
+
+                if (listener != null) {
+                    try {
+                        updateReadingStates(listener.getCurrentReading());
+                    } catch (Throwable th) {
+                        logger.error("Uncaught throwable in JeeLink sensor handler for thing " + getThing().getLabel()
+                                + " (" + getThing().getUID() + ")...", th);
+                    }
+                }
+            }
+        };
+        logger.debug("JeeLink sensor handler value propagation job for thing {} ({}) created with interval {} s",
+                getThing().getLabel(), getThing().getUID(), updateInterval);
+        return scheduler.scheduleAtFixedRate(runnable, updateInterval, updateInterval, TimeUnit.SECONDS);
+    }
+
+    private ScheduledFuture<?> createStatusUpdateJob() {
+        final int sensorTimeout = getConfig().as(JeeLinkSensorConfig.class).sensorTimeout;
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                if (listener != null && getThing().getStatus() == ThingStatus.ONLINE
+                        && (listener.getLastReadingTime() == -1
+                                || System.currentTimeMillis() - listener.getLastReadingTime() > sensorTimeout * 1000)) {
+                    logger.debug("Setting JeeLink sensor handler status for thing {} ({}) to OFFLINE",
+                            getThing().getLabel(), getThing().getUID());
+                    updateStatus(ThingStatus.OFFLINE);
+                }
+            }
+        };
+        logger.debug("JeeLink sensor timeout job for thing {} ({}) created with interval {} s", getThing().getLabel(),
+                getThing().getUID(), sensorTimeout);
+        return scheduler.scheduleAtFixedRate(runnable, sensorTimeout, 1, TimeUnit.SECONDS);
```

Comment:
Changed to scheduleWithFixedDelay. As you can see, it only checks the status, computes a difference and does a comparison. This should be as light-weight as it gets.